src/main/java/org/opensearch/knn/index/codec/derivedsource/DerivedSourceStoredFieldVisitor.java

```diff
@@ -15,14 +17,21 @@
 /**
  * Custom {@link StoredFieldVisitor} that wraps an upstream delegate visitor in order to transparently inject derived
  * source vector fields into the document. After the source is modified, it is forwarded to the delegate.
+ * Uses Lombok's {@code @Delegate} annotation to automatically delegate all non-overridden methods to the wrapped visitor.
  */
 @AllArgsConstructor
 public class DerivedSourceStoredFieldVisitor extends StoredFieldVisitor {
 
+    @Delegate
     private final StoredFieldVisitor delegate;
     private final Integer documentId;
     private final DerivedSourceVectorTransformer derivedSourceVectorTransformer;
 
+    @Override
+    public void binaryField(final FieldInfo fieldInfo, final StoredFieldDataInput value) throws IOException {
+        super.binaryField(fieldInfo, value);
+    }
```

Comment:
if we are using Delegate on `StoredFieldVisitor` then why we need this function?